src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelManager.cs

```diff
@@ -101,9 +130,14 @@ public IEnumerable<IViewModel> ActiveViewModels
         {
             get
             {
-                lock (_managedViewModelsLock)
+                _managedViewModelsLock.EnterReadLock();
```

Comment:
We could consider something like this instead, saves us a lot of try / finally:

```
using (new LockManagedViewModels())
{
    return GetAllViewModels(_managedViewModels).ToList(); 
}
```

The class can be like this:

```
public IDisposable LockManagedViewModels()
{
    return new DisposableToken<object>(this, x => 
    { 
        _managedViewModelsLock.EnterReadLock(); 
    }, x => 
    { 
        _managedViewModelsLock.ExitReadLock(); 
    });
}
```